teps/0092-scheduling-timeout.md

```diff
@@ -17,361 +24,474 @@ authors:
-  use cases as well which have nothing to do with load.
-* Kubernetes itself doesn't provide timeout like functionality by default; the closest feature is that
-  [failed and succeeded pods will be garbage collected](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-garbage-collection)
-  if the number of pods in existence exceeds a threshold.
+### Execution Timeout
 
-### Risks and Mitigations
+The `execution` timeout of a `TaskRun` applies from the scheduled time of its `Pod` until the completion time of `TaskRun`. The `execution` timeout has no default value.
 
-* This may open the door for even more fine-grained timeout requests, for example, not including the time to pull an
-  image in the timeout of the Run execution. This would technically be possible (e.g. use the difference between the
-  [`waiting` and `running` state of each container](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-state-waiting))
-  but if users want really fine grained control they may be better off implementing this in a separate controller
-  which can observe execution and apply policies. On the other hand, we already have the precedent of the `timeout`
-  section in our PipelineRun spec, maybe adding more configuration here won't be the end of the world.
-  * Mitigation: Consider this the last timeout related feature we add; on the next timeout related feature request
-    we consider building a separate controller to control timeouts. Not sure this is realistic though; maybe in reality
-    we're likely to end up accepting a number of different timeouts to support different scenarios.
+### Total Timeout
 
-### User Experience
+The `total` timeout of a `TaskRun` is the combination of `scheduling` and `execution` timeouts. It applies from the start time of the `TaskRun` until the completion time of the `TaskRun`. This timeout is useful when a user doesn't care to distinguish between scheduling vs execution, they only want to general timeout of a `TaskRun`.
 
-* As we add more timeout options it may become more difficult to reason about which timeout is applying when
-* Can't easily adapt this configuration to other requests (e.g. including or not including image pull time in the
-  timeout)
-* At the `PipelineRun` level, this timeout applies to individual `TaskRuns` which is not the case for any other
-  timeouts
+Defaults for `total` timeout:
+- If either `execution` or `scheduling` timeout is provided, the `total` timeout will default to `60` mins
+- Otherwise, the default `total` timeout will be calculated from `scheduling` + `execution`.
+* Note that this is different from how `pipelineRun.spec.timeouts` works: It's possible to declare PipelineRun timeouts where `timeouts.tasks` + `timeouts.finally` < `timeouts.pipeline`, and `timeouts.pipeline` always defaults to 60m and is never calculated by adding `timeouts.tasks` and `timeouts.finally`.
```

Comment:
I am not sure we need that note here. It does confuse me more than anything, as we are talking about `Task` and `TaskRun` here, so talking about `PipelineRun` seems out of place 😅